gdal/frmts/nitf/nitffile.c

```diff
@@ -2216,7 +2217,77 @@ static char** NITFGenericMetadataReadTREInternal(char **papszMD,
                 (*pnMDSize) ++;
                 papszTmp[0] = NULL;
                 CPLFree(papszTmp);
+                const char* pszVal = strchr(papszMD[(*pnMDSize) - 1], '=') + 1;
+                
+				if (strcmp(pszType, "IEEE754") == 0)
+				{
+                    int len = strlen(pszVal);
+                    unsigned char *cVal = pszVal;
+                    unsigned char ncVal[4] = {0x0,0x0,0x0,0x0};
+                    int i;
+                    for (i = 0; i< len; i++)
+                    {
+                        ncVal[i] = cVal[i];
+                    }
+                    /*https://bytes.com/topic/c/answers/212667-convert-unsigned-char-array-float*/
+                    unsigned int nVal = ((unsigned int)ncVal[0]<<24) |
+                        ((unsigned int)ncVal[1]<<16) |
+                        ((unsigned int)ncVal[2]<<8) |
+                        ((unsigned int)ncVal[3]);
+                    
+                    int s,e;
+                    long f;
+                    s = (nVal & 0x80000000UL) >> 31;
+                    e = (nVal & 0x7F800000UL) >> 23;
+                    f = (nVal & 0x007FFFFFUL);
+                    if (e == 255 && f != 0) {
+                        /* NaN (Not a Number) */
+                        sprintf(pszVal, "%f",0.0/0.0);
```

Comment:
overwrite pszVal is wrong here as it is a read-only variable. Use a temporary buffer

Reply:
Not sure how to make this change in order to make the correct values appear.